src/configs/encoders.py

```diff
@@ -0,0 +1,198 @@
+  attention_dropout_rate: float = 0.1
+  max_position_embeddings: int = 512
+  type_vocab_size: int = 2
+  initializer_range: float = 0.02
+  embedding_size: Optional[int] = None
+  output_range: Optional[int] = None
+  return_all_encoder_outputs: bool = False
+  # Pre/Post-LN Transformer
+  norm_first: bool = False
+
+
+@dataclasses.dataclass
+class MmtEncoderConfig(hyperparams.Config):
+  """Mmt encoder configuration."""
+
+  # Vocabulary size of `token_ids`.
+  vocab_size: int = 30522
+
+  # Vocabulary size of `segment_ids`.
+  segment_vocab_size: int = 16
+
+  # Size of `token_ids` embeddings.
+  # The default of `None` makes this equal to `hidden_size` like original
+  # BERT, but it can be set to a smaller value (e.g. 128) like ALBERT. Must
+  # be positive.
+  embedding_size: int = None
+
+  # Size of the encoder layers and the pooler layer.
+  hidden_size: int = 768
+
+  # Number of hidden layers in the Transformer encoder.
+  num_hidden_layers: int = 12
+
+  # Number of attention heads for each attention layer in the Transformer
+  # encoder
+  num_attention_heads: int = 12
+
+  # Maximum distance to use for relative position representations.
+  # All larger distances will be clipped to this value.
+  relative_pos_max_distance: int = 12
+
+  # The total vocabulary size for relative positions.
+  # This must be at least `2 * relative_pos_max_distance + 1`.
+  relative_vocab_size: int = 32
+
+  # If larger than 0, use `MmtRelativePositionGenerator` instead of
+  # `RelativeTransformerSideInputs`.
+  relative_att_num_core_layers: int = 0
+
+  # The maximum sequence length that this model might ever be used with;
+  # used for absolute position embeddings like BERT.
+  # If set to 0 (the default), we skip absolute position embeddings entirely.
+  # If nonzero, inputs larger than this value are not allowed.
+  max_absolute_position_embeddings: int = None
+
+  # The size of the "intermediate" (i.e., feed-forward) layer in the 
+  # Transformer encoder.
+  intermediate_size: int = 3072
+
+  # The non-linear activation function (function or string) in the encoder
+  # and pooler.
+  hidden_activation: str = 'gelu'
+
+  #  The dropout probability for all fully connected layers in the embeddings,
+  # encoder, and pooler.
+  hidden_dropout_prob: float = 0.1
+
+  # The dropout ratio for the attention probabilities.
+  attention_probs_dropout_prob: float = 0.1
+
+  # The stdev of the truncated_normal_initializer for initializing all weight
+  # matrices.
+  initializer_range: float = 0.02
+
+  # If True, use "pre-activation" order for residual blocks.
+  use_pre_activation_order: bool = False
+
+  # If True, use one-hot lookup for embedding layers with samller vocab.
+  use_one_hot_lookup: bool = True
+
+  # If True, use pooler layer to get CLS token embedding.
+  use_pooler_layer: bool = False
+
+
+@dataclasses.dataclass
+class EncoderConfig(hyperparams.OneOfConfig):
+  """Encoder configuration."""
+  type: Optional[str] = "bert"
```

Comment:
Maybe these should be declared as global constants, to prevent typos causing unexpected behavior (e.g., if instead of "mmt", you were to pass "MMT", it will use the BERT configuration.)